modules/imap/handler_modules.php

```diff
@@ -633,7 +633,7 @@ public function process() {
                 $this->out('imap_expanded_folder_path', $path);
             }
             else {
-                Hm_Msgs::add(sprintf('ERRCould not authenticate to the selected %s server (%s)', $imap->server_type, $form['imap_user']));
+                Hm_Msgs::add(sprintf('ERRCould not authenticate to the selected %s server (%s)', $imap->server_type, get_failed_email($imap->show_debug())));
```

Comment:
This solves the problem however it creates a direct dependency between a string generated by the IMAP Debugger and the listing method, wich can eventually cause other problems in the future.

You can get the server informations directly from the ID that was sent at the beginning of the method using the module's `user_config`:

```
$this->user_config->get('imap_servers')[$form['imap_server_id']]
```

Reply:
Thank you @henrique-borba, I solve it using `user_config` module